### ISSUE TITLE
Introduce range-over-func `iter.Seq` method `All()` for `graph.EndpointPair`

### ISSUE THREAD

**Issue by jbduncan** [jbduncan/go-containers#7]
Blocked on the currently-experimental [range-over-func](https://github.com/golang/go/issues/61405) Go feature being merged into a stable version of Go.

This will allow the nodes of an endpoint pair to be iterated on with `for` loops, which is more flexible than introducing a `ForEach` method which may not allow early exits from the loop.

**Comment by jbduncan** [comment]
For clarity, this would treat `graph.EndpointPair` as a two-element collection with a source and target node that the iterator would return.

**Comment by jbduncan** [comment]
Done.
